src/transformers/pipelines/token_classification.py

```diff
@@ -210,63 +244,124 @@ def __call__(self, inputs: Union[str, List[str]], **kwargs):
         _inputs, offset_mapping = self._args_parser(inputs, **kwargs)
         if offset_mapping:
             kwargs["offset_mapping"] = offset_mapping
-
         return super().__call__(inputs, **kwargs)
 
-    def preprocess(self, sentence, offset_mapping=None):
+    def preprocess(self, sentence, offset_mapping=None, **preprocess_params):
+        tokenizer_params = preprocess_params.pop("tokenizer_params", {})
         truncation = True if self.tokenizer.model_max_length and self.tokenizer.model_max_length > 0 else False
-        model_inputs = self.tokenizer(
+        inputs = self.tokenizer(
             sentence,
             return_tensors=self.framework,
             truncation=truncation,
             return_special_tokens_mask=True,
             return_offsets_mapping=self.tokenizer.is_fast,
+            **tokenizer_params,
         )
-        if offset_mapping:
-            model_inputs["offset_mapping"] = offset_mapping
+        num_chunks = len(inputs["input_ids"])
 
-        model_inputs["sentence"] = sentence
+        for i in range(num_chunks):
+            if self.framework == "tf":
+                model_inputs = {k: tf.expand_dims(v[i], 0) for k, v in inputs.items()}
+            else:
+                model_inputs = {k: v[i].unsqueeze(0) for k, v in inputs.items()}
+            if offset_mapping:
+                model_inputs["offset_mapping"] = offset_mapping if i == 0 else None
+            model_inputs["sentence"] = sentence if i == 0 else None
+            model_inputs["is_last"] = i == num_chunks - 1
 
-        return model_inputs
+            yield model_inputs
 
     def _forward(self, model_inputs):
         # Forward
-        special_tokens_mask = model_inputs.pop("special_tokens_mask")
-        offset_mapping = model_inputs.pop("offset_mapping", None)
-        sentence = model_inputs.pop("sentence")
+        model_outputs = model_inputs
+        model_inputs = {k: model_inputs[k] for k in self.tokenizer.model_input_names}
```

Comment:
No the previous code was selectively popping `offset_mapping` and `sentence`. Which is not the same as this, which relies on `self.tokenizer.model_input_names` to be correct. 

The issue with that approach, is that should `model_input_names` fail to mention `attention_mask` for instance, then it would silently start to behave incorrectly (by failing to mask the proper input_ids).

`model.forward(**tokenizer(string))`  is the invariant we should upkeep.

Since we have to add more than necessary for the post processing, we are in control of which to remove too in the pipeline.
That way we don't have on tokenizer internals (which could and unfortunatly sometimes are incorrect).

Do you see the difference now ?

Reply:
Yes I understand, here we rely only on the tokenizer. I'll revert back to the previous function. (This function dosn't need to be modified)